### PR TITLE
[20/N] Fix clang-tidy warnings in jit 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #133067

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

File: torch/csrc/jit/cuda/cuda.h [hunk 1]
```diff
@@ -22,7 +21,6 @@ class CUDAStream final : public CustomClassHolder {
         c10::cuda::getStreamFromPool(static_cast<int>(priority), device_index));
   }
 
-  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
   CUDAStream(c10::cuda::CUDAStream s) {
```
Comment (Skylion007) [inline torch/csrc/jit/cuda/cuda.h:24]: Hmm, CUDAStream is trivially copyable?

Comment (cyyever) [reply torch/csrc/jit/cuda/cuda.h:24]: This lint is about member initialisation

**Comment by cyyever** [comment]
@Skylion007 The CI failures seem unrelated

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/jit/codegen/fuser/tensor_info.h [hunk 2]
```diff
@@ -18,12 +17,8 @@ struct TORCH_API TensorInfo {
   }
 
   void* data;
-#pragma GCC diagnostic ignored "-Wpedantic"
```
Comment (Skylion007) [inline torch/csrc/jit/codegen/fuser/tensor_info.h:21]: Hmm, we sure we don't need this anymore? I am not sure if we have pedantic enabled generally throughout the codebase so I'd leave it in.

Comment (cyyever) [reply torch/csrc/jit/codegen/fuser/tensor_info.h:21]: Enabled globally in some previous PR.

**Pull request closed without merging**